packages/nimbus/src/components/button/button.mdx

```diff
@@ -17,153 +17,767 @@ figmaLink: >-
+      <li>
+        Use the correct and consistently patterned icon for the action that a
+        user will be selecting.{" "}
+      </li>
+      <li>
+        Ok ways to break this rule are with drop downs and right pointing
+        arrows.
+      </li>
+    </ul>
+  </div>
+</Tip>
+<br />
+
+```jsx-live
+const App = () => (
+  <Stack direction="horizontal">
+    <Button tone="primary" size="md" variant="solid">
+      <Icons.Undo />
+      Revert
+    </Button>
+    <Button tone="primary" size="md" variant="solid">
+      <Icons.FileUpload />
+      Upload
+   </Button>
+    <Button tone="primary" size="md" variant="solid">
+      <Icons.Add />
+      Add product
+   </Button>
+    <Button tone="primary" size="md" variant="outline">
+      <Icons.ArrowBack />
+      Previous
+   </Button>
+    <Button tone="primary" size="md" variant="solid">
+      Next
+      <Icons.ArrowForward />
+   </Button>
+ </Stack>
+)
 ```
 
-### Icons
+<Tip tipColor="#E5484D" tipBackgroundColor="#FFF5F5" icon={<Icons.Error />}>
+  <div>
+    <strong>Don't</strong>
+    <ul style={{ listStyle: "disc", marginLeft: "20px" }}>
+      <li>Icons should represent the action the user will be taking.</li>
+      <li>Don’t use two icons to represent an action. One icon will do.</li>
+      <li>Left icons are the default unless there is directional reason.</li>
+    </ul>
+  </div>
+</Tip>
+<br />
+
+```jsx-live
+const App = () => (
+  <Stack direction="horizontal">
+    <Button tone="primary" size="md" variant="solid">
+      <Icons.Upload />
+       Delete
+      <Icons.Upload />
+    </Button>
+    <Button tone="critical" size="md" variant="solid">
+      <Icons.Clear />
+        Add discount
+      <Icons.Clear />
+    </Button>
+    <Button tone="primary" size="md" variant="solid">
+      Revert
+      <Icons.Undo />
+    </Button>
+ </Stack>
+)
+```
 
-Simply drop the desired icon into the `Button`:
+<Tip tipColor="#46A557" tipBackgroundColor="#F4FAF5" icon={<Icons.Lightbulb />}>
+  <div>
+    <strong>Do</strong>
+    <ul style={{ listStyle: "disc", marginLeft: "20px" }}>
+      <li>
+        Use right icons to build a drop down component with the Popover
+        component.
+      </li>
+      <li>
+        Icons are on the left unless there is a directional reason to use a
+        right icon placement.
+      </li>
+    </ul>
+  </div>
+</Tip>
+<br />
 
 ```jsx-live
-const App = () => {
-  const sizes = ["2xs", "xs", "md"].reverse();
+const App = () => (
+  <Stack direction="horizontal">
+    <Stack>
+      <Button tone="primary" size="md" variant="outline">
+        Save changes
+        <Icons.KeyboardArrowDown />
+      </Button>
+     <Card.Root elevation="elevated" cardPadding="md">
+        <Card.Content>
+        <div style={{fontSize: "16px", fontWeight: "400"}}>
+        <p>Save</p>
+        <p>Save and publish</p>
+        <p>Save and open next</p>
+        </div>
+        </Card.Content>
+      </Card.Root>
```

Comment:
This isn't actually how we're doing buttons with dropdowns, right? 😅

Reply:
Nope. Otherwise, we would just use the select in the snippet instead. This is mainly for the visuals and we may need to look into how to turn off the snippets for cases like this.

Reply:
This is perhaps a stupid question. Is this documentation appropriate for this location? Are you saying that someone could never have a dropdown open from a button? This particular question does not need to hold up this PR.